sdk/core/azure-core/src/main/java/com/azure/core/credential/KeyCredential.java

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.credential;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Objects;
+
+/**
+ * Represents a credential that uses a key to authenticate.
+ */
+public class KeyCredential {
```

Comment:
We should finalize the name with architects. The proposal in arch board was to call this `RotatingKeyCredential`. Although, my preference is to call this `KeyCredential`.

Reply:
I get @johanste's breath to like the name offline.

Reply:
Name was discussed again in the OpenAI archboard and all architects agreed to use this name.